### ISSUE TITLE
non-DXF ADD INDEX of large table (trigger import due to disk quota) will write KV with TS = 0

### ISSUE THREAD

**Issue by lance6716** [pingcap/tidb#57980]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

use this diff to trigger disk quota import on small tables

```diff
diff --git a/pkg/ddl/ingest/backend.go b/pkg/ddl/ingest/backend.go
index f11849f70b..cd08ec14c5 100644
--- a/pkg/ddl/ingest/backend.go
+++ b/pkg/ddl/ingest/backend.go
@@ -187,6 +187,8 @@ func (bc *litBackendCtx) Flush(indexID int64, mode FlushMode) (flushed, imported
        }

        shouldFlush, shouldImport := bc.checkFlush(mode)
+       shouldFlush = true
+       shouldImport = true
        if !shouldFlush {
                return false, false, nil
        }
```

and print TS used in import, or check TiKV

run `TestAddIndexRemoteDuplicateCheck` with the change, or other ADD INDEX that can trigger disk-quota import

```diff
diff --git a/tests/realtikvtest/addindextest4/ingest_test.go b/tests/realtikvtes
t/addindextest4/ingest_test.go
index 2e393ce4f8..7a3f91f031 100644
--- a/tests/realtikvtest/addindextest4/ingest_test.go
+++ b/tests/realtikvtest/addindextest4/ingest_test.go
@@ -407,7 +407,8 @@ func TestAddIndexRemoteDuplicateCheck(t *testing.T) {
        tk.MustExec("insert into t values(100000, 1, 1);")

        ingest.ForceSyncFlagForTest = true
-       tk.MustGetErrCode("alter table t add unique index idx(b);", errno.ErrDupEntry)
+       tk.MustExec("alter table t add index idx_test(b);")
+       //tk.MustGetErrCode("alter table t add unique index idx(b);", errno.ErrDupEntry)
        ingest.ForceSyncFlagForTest = false

        tk.MustExec("set global tidb_ddl_reorg_worker_cnt=4;")
```

### 2. What did you expect to see? (Required)

no error

### 3. What did you see instead (Required)

TiKV panic due to it has data with TS = 0

**Comment by lance6716** [comment]
introduced by https://github.com/pingcap/tidb/pull/52993 https://github.com/pingcap/tidb/pull/53188

**Comment by tangenta** [comment]
Trigger conditions:

1. Run create-index DDL.
1. The target index size exceeds the 'temp-dir' capacity and `@@tidb_ddl_disk_quota`(by default, it is 100GiB).
1. `@@tidb_enable_dist_task` is disabled.

**Comment by AndreMouche** [comment]
summary and workaround for this issue:

TiKV frequent panic Due to committed version with `start_ts=0`   [tidb#57980](https://github.com/pingcap/tidb/issues/57980) 
- Phenomenon: When `tidb_ddl_enable_fast_reorg` is enabled, but `tidb_enable_dist_task` is not. If TiDB generates a large number of SST files and upload them to TiKV(especially for very large tables), the batch generation may not finish all SSts at once. During this process,start_ts may be left un initialized, causing some SST files to have `start_ts=0`
  - In version v8.1.1, `tidb_enable_dist_task` is enabled by default, but clusters upgraded from older versions may have it set to false
  - Panics mainly occur in
    - `get_keys` in TiKV GC worker
    - Prewrite checks for the most recent committed versions
- Trigger Conditions: This may occur when all of the following conditions are met:
  - tidb_ddl_enable_fast_reorg is enabled
  - tidb_enable_dist_task is disabled
  - Adding an index on a large table 
- Workaround:
  - STOP the related  DDL job if needed
ADMIN CANCEL DDL JOBS job_id [, job_id] ...;
  - Identify the problematic indexes, drop them
  - Wait for next GC, if gc stucks, try to recover it. After the GC lifetime, no panic should happens
  - Disable DDL reorg
set global `tidb_ddl_enable_fast_reorg=OFF`   
  - If `tidb_ddl_enable_fast_reorg` is enabled, ensure `tidb_enable_dist_task` is also enabled
- Fixed version: v8.1.2
